aten/src/ATen/native/Normalization.cpp

```diff
@@ -176,39 +176,42 @@ std::tuple<Tensor,Tensor,Tensor> batch_norm_cpu_transform_input_template(
     return std::make_tuple(output, save_mean, save_invstd);
   }
 
-  Tensor output = at::empty_like(input, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
-
-  int64_t n_input = input.size(1);
-
-  auto save_mean_a = conditional_accessor_1d<scalar_t>(save_mean);
-  auto save_invstd_a = conditional_accessor_1d<scalar_t>(save_invstd);
-
-  auto running_mean_a = conditional_accessor_1d<scalar_t>(running_mean);
-  auto running_var_a = conditional_accessor_1d<scalar_t>(running_var);
-
-  parallel_for(0, n_input, 1, [&](int64_t b_begin, int64_t b_end) {
-    for (int64_t f = b_begin; f < b_end; ++f) {
-      Tensor in = input.select(1, f);
-      Tensor out = output.select(1, f);
-
-      scalar_t mean, invstd;
-      if (train) {
-        mean = save_mean_a[f];
-        invstd = save_invstd_a[f];
-      } else {
-        mean = running_mean_a[f];
-        invstd = 1 / std::sqrt(running_var_a[f] + eps);
-      }
-
-      // compute output
-      scalar_t w = weight.defined() ? weight.data_ptr<scalar_t>()[f * weight.stride(0)] : 1;
-      scalar_t b = bias.defined() ? bias.data_ptr<scalar_t>()[f * bias.stride(0)] : 0;
-
-      auto iter = TensorIterator::unary_op(out, in);
-      cpu_serial_kernel(iter, [=](const scalar_t i) -> scalar_t {
-        return ((i - mean) * invstd) * w + b;
-      });
+  const int64_t ndim = input.dim();
+  // Helper to convert 1d tensors to an nd tensor that broadcasts with input
+  // All elements go into the channel dimension
+  DimVector sizes(ndim, 1), strides(ndim, 0);
+  auto as_nd = [&](const Tensor& t) {
+    TORCH_INTERNAL_ASSERT(t.defined() && t.dim() == 1);
+    sizes[1] = t.sizes()[0];
```

Comment:
strictly speaking you need to be sure that ndim is >0 here, but I trust there was some check to that effect before batch_norm_cpu_transform_input_template was called